cluster/charts/crossplane/templates/rbac-manager-clusterrole.yaml

```diff
@@ -41,6 +41,12 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
+  - pkg.crossplane.io
+  resources:
+  - providerrevisions/finalizers
+  verbs:
+  - update
```

Comment:
Could you add a comment here please. Something like:

> The RBAC manager creates an RBAC ClusterRole for each ProviderRevision it sees. It needs RBAC access to update the finalizers of ProviderRevisions in order to set blockOwnerDeletion on ClusterRoles owned by a ProviderRevision per https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement

Reply:
Also, I think a few other updates are needed to cover the case where the RBAC manager creates RBAC roles that block the deletion of their owners:

* https://github.com/crossplane/crossplane/blob/377239/internal/controller/rbac/provider/roles/roles.go#L145
* https://github.com/crossplane/crossplane/blob/377239/internal/controller/rbac/definition/roles.go#L168
* https://github.com/crossplane/crossplane/blob/377239/internal/controller/rbac/namespace/roles.go#L84

You covered the first one `providerrevisions/finalizers`, but I think it also needs `namespaces/finalizers` and `compositeresourcedefinitions/finalizers`.